### ISSUE TITLE
FAILED TEST: D-Bus Socket Options

### ISSUE THREAD

**Issue by frkli1965** [bus1/dbus-broker#325]
Hi,

Been having problems with the latest test case as follows:-

=================================== 48/48 ====================================
test:         dbus-broker / D-Bus Socket Options
start time:   13:36:47
duration:     0.20s
result:       killed by signal 6 SIGABRT
command:      MALLOC_PERTURB_=119 /mnt/Media/GIT/Builds/dbus-broker/build/src/test-sockopt

----------------------------------- stderr -----------------------------------
ERROR sockopt_get_peerpidfd @ ../src/util/sockopt.c +242: Invalid argument
test-sockopt: ../src/util/test-sockopt.c:86: test_peerpidfd_client: Assertion `false && "!r"' failed.
test-sockopt: ../src/util/test-sockopt.c:23: wait_and_verify: Assertion `false && "WIFEXITED(status)"' failed.
==============================================================================

Summary of Failures:

48/48 dbus-broker / D-Bus Socket Options         FAIL            0.20s   killed by signal 6 SIGABRT

Ok:                            47  
Expected Fail:            0   
Fail:                             1   
Unexpected Pass:     0   
Skipped:                     0   
Timeout:                    0

Linux Kernel: v6.5rc5
CPU: AMD 5600x

Meson Build Options:  -Dapparmor=true -Daudit=true -Ddocs=true -Dlinux-4-17=true -Dselinux=true

Should I be concerned or just ignore it?

Kind regards,

**Comment by dvdhrm** [comment]
No need to be concerned. It tests new behavior of to-be-released linux-6.5 and upstream discussion is ongoing. I will disable the test once back home.

Just curious: Do you test dbus-broker git on a daily basis, or how did you notice?

**Comment by dvdhrm** [comment]
(Closing as it should be fixed on `main`)

**Comment by frkli1965** [comment]
Hi David,

Thanks for getting back to me, much appreciated...

Regarding your question, I literally run GIT OS these days.

What started out as Ubuntu-Mate in the dim and distant past has essentially morphed into an OS whose main constituents e.g. Linux kernel, systemd, dbus, dbu-broker, binutils, coreutils, wayland, gtk4, zlib-ng, SSL, SSH, gnutls, apparmor, polkit, pipewire, alsa, bash, Vulkan, Hyprland, Sway, etc. are all now compiled from source whenever there are new commits with the aid of an semi-automated build system that I concocted.

Initially a little experiment has basically transitioned into a whole new Linux distro in its own right comprising of what is now well over 150 core components that are regularly cloned from Github, Gitlab et al.

A kind of 'Linux From Scratch' but in reverse I suppose and needless to say, your excellent dbus-broker performs an fundamental role in keeping things ship-shape and tiggerty-boo.

Ah, but doesn't that make for a rather unstable system one might rightly ask? Well, much to my pleasant surprise, it is remarkably stable as a matter of fact. To date, I have had less trouble with it than any other distro I cared to hop onto in the past tbh.

Anyway, hope I haven't bored you too much, though you did ask... :-)

Going forward, I intend to move away from Canonical despite Ubuntu having been a most resilient base, as I'm not entirely sure of their future plans i.e. snaps and all that and moving onto Mageia OS or something similar, as I hope to turn my mad-cap experiment into a rolling distro in due course...

All the best...

PS. Many thanks for resolving the issue so expeditiously and keep up the great work as always.
